editor/src/components/canvas/canvas-strategies/strategies/draw-to-insert-metastrategy.tsx

```diff
@@ -48,22 +51,88 @@ import {
+import { getApplicableReparentFactories } from './reparent-metastrategy'
+import { ReparentStrategy } from './reparent-strategy-helpers'
 
-export function drawToInsertStrategy(
+export const drawToInsertMetaStrategy: MetaCanvasStrategy = (
   canvasState: InteractionCanvasState,
   interactionSession: InteractionSession | null,
   customStrategyState: CustomStrategyState,
+): Array<CanvasStrategy> => {
+  if (
+    interactionSession == null ||
+    !(
+      interactionSession.interactionData.type === 'DRAG' ||
+      interactionSession.interactionData.type === 'HOVER'
+    )
+  ) {
+    return []
+  }
+  const pointOnCanvas =
+    interactionSession.interactionData.type === 'DRAG'
+      ? interactionSession.interactionData.originalDragStart
+      : interactionSession.interactionData.point
+
+  const cmdPressed = interactionSession.interactionData.modifiers.cmd
+
+  const applicableReparentFactories = getApplicableReparentFactories(
+    canvasState,
+    pointOnCanvas,
+    cmdPressed,
+    true, // <- TODO this is an important assumption, make sure this is still true when inserting into a flex storyboard
+  )
+
+  return mapDropNulls((result): CanvasStrategy | null => {
+    const name = getDrawToInsertStrategyName(result.strategyType, result.missingBoundsHandling)
+
+    return drawToInsertStrategyFactory(
+      canvasState,
+      interactionSession,
+      customStrategyState,
+      result.factory,
+      name,
+      result.fitness,
+      result.targetParent,
+      result.targetIndex,
+    )
+  }, applicableReparentFactories)
+}
+
+function getDrawToInsertStrategyName(
+  strategyType: ReparentStrategy,
+  missingBoundsHandling: MissingBoundsHandling,
+) {
+  switch (strategyType) {
+    case 'REPARENT_TO_ABSOLUTE':
+      if (missingBoundsHandling === 'use-strict-bounds') {
+        return 'Draw to Insert (Abs)'
+      } else {
+        return 'Draw to Insert (Abs, Forced)'
+      }
+    case 'REPARENT_TO_FLEX':
+      return 'Draw to Insert (Flex)'
+  }
+}
+
+function drawToInsertStrategyFactory(
+  canvasState: InteractionCanvasState,
+  interactionSession: InteractionSession | null,
+  customStrategyState: CustomStrategyState,
+  reparentStrategyToUse: CanvasStrategyFactory,
+  name: string,
+  fitness: number,
+  targetParent: ElementPath | null,
+  targetIndex: number | null,
 ): CanvasStrategy | null {
   const insertionSubjects = getInsertionSubjectsFromInteractionTarget(canvasState.interactionTarget)
   if (insertionSubjects.length !== 1) {
     return null
   }
   const insertionSubject = insertionSubjects[0]
   return {
-    id: 'DRAW_TO_INSERT',
-    name: 'Draw to insert',
+    id: name, // TODO review this before merge
+    name: name,
     controlsToRender: [
       // TODO the controlsToRender should instead use the controls of the actual canvas strategy -> to achieve that, this should be a function of the StrategyState here
       controlWithProps({
```

Comment:
You can update this to provide the `targetParent` to `ParentOutlines` and `ParentBounds` after #2644 is merged

Reply:
done!